client/ayon_core/hosts/nuke/plugins/publish/collect_headless_farm.py

```diff
@@ -0,0 +1,42 @@
+import pyblish.api
+
+
+class CollectHeadlessFarm(pyblish.api.ContextPlugin):
+    """Setup instances for headless farm submission."""
+
+    order = pyblish.api.CollectorOrder + 0.4999
+    label = "Collect Headless Farm"
+    hosts = ["nuke"]
+
+    def process(self, context):
+        if not context.data.get("headless_farm", False):
+            return
+
+        for instance in context:
+            if instance.data["family"] == "workfile":
+                instance.data["active"] = False
+
+                # Disable version validation.
+                instance.data.pop("latestVersion")
+                continue
+
+            # Filter out all other instances.
+            node = instance.data["transientData"]["node"]
+            if node.name() != instance.context.data["node_name"]:
+                instance.data["active"] = False
+                continue
+
+            # Enable for farm publishing.
+            instance.data["farm"] = True
+
+            # Clear the families as we only want the main family, ei. no review
+            # etc.
+            instance.data["families"] = []
```

Comment:
I would add a family here so the instance can be recognized.

Why:
- Change of `"NukeSubmitDeadline"` should not happen in this plugin, it would actually crash if you don't have deadline addon enabled, should be moved to deadline plugin instead (with instance families filtering). Why we're actually changing "settings", there is no way how to tell to not use published workfile?
- The same applies to `ExtractHeadlessFarm` where should be families filtering.

Reply:
Done